File: Julia/LLMapReduce/word_count/top5overall_map.jl
# Activate environment

include("../../word_count_helpers.jl")
using FileIO,JSON

# Grab the two arguments that are passed in
# The first is the input file name and the second is the output file name
#inputfile = ARGS[1]
#outputfile = ARGS[2]
inOutFileList = readlines(open(ARGS[1],"r+"))
print(inOutFileList)

# Read in file and clean the text
for line in inOutFileList
    inputfile = split(line)[1]
    outputfile = split(line)[2]

    # Read in file and clean the text
    f = open(inputfile)
    text=cleantext(readlines(f))

    # Count number of times each word appears
    wordCounts = countwords(text)

    # Save the word counts to the passed in output file name
    open(outputfile*".json","w") do f
    	JSON.print(f,wordCounts)
    end
end

File: Python/word_count/LLMapReduce/top5overall_map.py
import os, sys, json
sys.path.append('../../')
from word_count_helpers import *

inOutFileList = open(sys.argv[1],"r+")

for line in inOutFileList.readlines():

    # Get input and output file names for this iteration
    (inFile,outFile) = line.split()
    print("Reading from " + inFile + " and writing to " + outFile)

    # Read in file and clean the text
    infid  = open(inFile,"r+", encoding='utf-8')
    text = cleantext(infid.readlines())

    # Count number of times each word appears
    wordCounts =countwords(text)

    # Save the word counts to the passed in output file name 
    with open(outFile+'.json', 'w') as fp:
        json.dump(wordCounts, fp)
